src/org/zaproxy/zap/utils/TableExportButton.java

```diff
@@ -0,0 +1,134 @@
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.zaproxy.zap.utils;
+
+import java.awt.event.ActionEvent;
+import java.io.File;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.util.ArrayList;
+import java.util.List;
+
+import javax.swing.AbstractAction;
+import javax.swing.ImageIcon;
+import javax.swing.JButton;
+import javax.swing.JOptionPane;
+import javax.swing.JTable;
+
+import org.apache.commons.csv.CSVFormat;
+import org.apache.commons.csv.CSVPrinter;
+import org.apache.log4j.Logger;
+import org.parosproxy.paros.Constant;
+import org.parosproxy.paros.view.View;
+import org.zaproxy.zap.view.widgets.WritableFileChooser;
+/**
+ * A {@code JButton} class to facilitate exporting tables to a file (such as CSV).
+ * 
+ * @since TODO add version
+ */
+public class TableExportButton extends JButton {
+
+	private static final long serialVersionUID = 3437613469695367668L;
+
+	private static final Logger LOGGER = Logger.getLogger(TableExportButton.class);
+	
+	private JTable exportTable = null;
+	
+	/**
+	 * Constructs a {@code TableExportButton} for the given table.
+	 * 
+	 * @param table the Table for which the data should be exported
+	 */
+	public TableExportButton(JTable table) {
+		super(Constant.messages.getString("export.button.name"));
+		setTable(table);
+		super.setIcon(new ImageIcon(TableExportButton.class.getResource("/resource/icon/16/115.png")));
+		super.addActionListener((new AbstractAction() {
+			private static final long serialVersionUID = 1L;
+
+			@Override
+			public void actionPerformed(ActionEvent e) {
+				WritableFileChooser chooser = new WritableFileChooser();
+				chooser.setSelectedFile(new File(Constant.messages.getString("export.button.default.filename")));
+				if (chooser.showSaveDialog(View.getSingleton().getMainFrame()) == WritableFileChooser.APPROVE_OPTION) {
+					String file = chooser.getSelectedFile().toString();
+					if (!file.endsWith(".csv")) {
+						file += ".csv";
+					}
+					try (CSVPrinter pw = new CSVPrinter(
+							Files.newBufferedWriter(chooser.getSelectedFile().toPath(), StandardCharsets.UTF_8),
+							CSVFormat.DEFAULT)) {
+						pw.printRecord(getColumnNames(getTable()));
+						int rowCount = getTable().getRowCount();
+						int colCount = getTable().getColumnCount();
+						for (int row = 0; row < rowCount; row++) {
+							List<Object> valueOfRow = new ArrayList<Object>();
+							for (int col = 0; col < colCount; col++) {
+								String value = getTable().getValueAt(row, col) == null ? "" : getTable().getValueAt(row, col).toString();

```

Comment:
Worth tweaking this to just call `getValueAt` once. Maybe delay the null check for the following statement?